runtime/str.go

```diff
@@ -711,8 +755,6 @@ func adjustIndex(i, l int) int {
 		i = 0
 	case i < 0:
 		i += l
-	case i > l:
-		i = l
```

Comment:
I was doing bound check on outside of that function, so I didn't notice that check was needed.

That case block has side effects,`"".startswith("", 1)` return True, but it should be False, when you run in real python.